From 715377367288a89cc64782b38ff0f6876cef58a7 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 16 May 2014 20:13:31 -0400
Subject: [md] dm-thin: allow metadata commit if pool is in PM_OUT_OF_DATA_SPACE mode

Message-id: <1400271213-25480-8-git-send-email-snitzer@redhat.com>
Patchwork-id: 79878
O-Subject: [RHEL7.1 PATCH 7/9] dm thin: allow metadata commit if pool is in PM_OUT_OF_DATA_SPACE mode
Bugzilla: 1091852
RH-Acked-by: Joe Thornber <thornber@redhat.com>
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>

BZ: 1091852

Upstream commit 8d07e8a5f5bc7b90f755d9b427ea930024f4c986
Author: Joe Thornber <ejt@redhat.com>
Date:   Tue May 6 16:28:14 2014 +0100

    dm thin: allow metadata commit if pool is in PM_OUT_OF_DATA_SPACE mode

    Commit 3e1a0699 ("dm thin: fix out of data space handling") introduced
    a regression in the metadata commit() method by returning an error if
    the pool is in PM_OUT_OF_DATA_SPACE mode.  This oversight caused a thin
    device to return errors even if the default queue_if_no_space ENOSPC
    handling mode is used.

    Fix commit() to only fail if pool is in PM_READ_ONLY or PM_FAIL mode.

    Reported-by: qindehua@163.com
    Signed-off-by: Joe Thornber <ejt@redhat.com>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>
    Cc: stable@vger.kernel.org # 3.14+

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-thin.c b/drivers/md/dm-thin.c
index 3259aff..fdaeb01 100644
--- a/drivers/md/dm-thin.c
+++ b/drivers/md/dm-thin.c
@@ -929,7 +929,7 @@ static int commit(struct pool *pool)
 {
  int r;
 
- if (get_pool_mode(pool) != PM_WRITE)
+ if (get_pool_mode(pool) >= PM_READ_ONLY)
   return -EINVAL;
 
  r = dm_pool_commit_metadata(pool->pmd);
-- 
1.7.1